### PR TITLE
remove unused states if state_retention_period is defined and > 0

### LINKED ISSUE #1639: Delete unused states
### Description

The goal would be for Taipy Gui to delete unused states so applications don't grow continuously.

This might be the cause of demos deployed by users or by us.

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1639

new config state_retention_period: the number of seconds to wait before removing a state after the last client is disconnected.

`tp.Gui(scenario_md).run(title="1639 remove unused states", state_retention_period=15)`

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]
Great that you are using the ws disconnect! Thanks for sorting the config alphabetically.

File: taipy/gui/_default_config.py [hunk 1]
```diff
@@ -58,12 +58,14 @@
...
     "notebook_proxy": True,
     "notification_duration": 3000,
+    "port": 5000,
     "propagate": True,
     "run_browser": True,
     "run_in_thread": False,
     "run_server": True,
     "server_config": None,
     "single_client": False,
+    "state_timeout": 0,
```
Comment (FabienLelaquais) [inline taipy/gui/_default_config.py]: "state_retention_period"?

**Review by FabienLelaquais — DISMISSED** [review]
Love it.
Not so much the setting name...

**Review by FabienLelaquais — APPROVED** [review]
👍

**Review by dinhlongviolin1 — APPROVED** [review]

**Pull request merged**